src/Umbraco.Web.UI.Client/src/common/services/focus.service.js

```diff
@@ -0,0 +1,72 @@
+/**
+ @ngdoc service
+ * @name umbraco.services.focusService
+ *
+ * @description
+ * <b>Added in Umbraco 8.1</b>. Application-wide service for focus related stuff.
+ * 
+ */
+
+(function () {
+    "use strict";
+
+    function focusService() {
+        
+        
+        function focusInApp(e) {
+            service.currentFocus = e.target;
+        }
+        document.addEventListener('focusin', focusInApp);
+        
+        
+        var service = {
+            currentFocus: null,
+            
+            /**
+    		 * @ngdoc property
+             * @name umbraco.services.focusService#lastKnownFocus
+		     * @propertyOf umbraco.services.focusService
+             *
+             * @description
+             * A element set to be remembered, the directive using this should store the value of this to make sure that its not changed white using that directive.
+             * This variable is avaiable for directives that are not able to figure out the focused element on init, and there this service will help remembering it untill the directive is initialized.
+             * 
+             */
+            lastKnownFocus: null,
```

Comment:
This seems dangerous to have a public get/set field here since it would be very easy to have a 2 way binding bound directly to this. Why not make these getter and setter methods instead to avoid the scenario you are mentioning in your comments? Same goes for the `currentFocus` field.

The storage mechanism behind these can be variables higher up in the closure.